src/java/org/apache/cassandra/net/AsyncStreamingOutputPlus.java

```diff
@@ -170,36 +170,50 @@ public int writeToChannel(Write write, StreamRateLimiter limiter) throws IOExcep
      */
     public long writeFileToChannel(FileChannel file, StreamRateLimiter limiter) throws IOException
     {
-        // write files in 1MiB chunks, since there may be blocking work performed to fetch it from disk,
-        // the data is never brought in process and is gated by the wire anyway
         if (channel.pipeline().get(SslHandler.class) != null)
-            return writeFileToChannel(file, limiter, 1 << 20, 1 << 20, 2 << 20);
+            // each batch is loaded into ByteBuffer, 64kb is more BufferPool friendly.
+            return writeFileToChannel(file, limiter, 1 << 16);
```

Comment:
@jasonstack Would it be less brittle here if we just made the batch size `BufferPool.NORMAL_CHUNK_SIZE`? That seems like it would still avoid passing the threshold where we have to allocate new buffers, and would adjust automatically if we ever had to change the normal chunk size. The thing I don't know is what the optimal buffer size actually is. If we want to stay as close as possible to the existing behavior, using `BufferPool.NORMAL_CHUNK_SIZE` obviously makes sense.

Reply:
Using 128kb (NORMAL_CHUNK_SIZE) won't cause out-of-pool allocation.

But current default low/high water mark is 32kb/64kb, I think a lower batch size will allow the write thread to wake up earlier and perform the blocking disk read earlier, assuming netty worker thread is flushing at a constant rate..

> The thing I don't know is what the optimal buffer size actually is.

My gut feeling is somewhere between low water mark and high water mark.. Legacy stream writer uses 64kb size. Zero-copy has higher batch-size and low/high water mark because buffer won't be copied to user space..

Reply:
Fair enough. I wonder if we could just make it `defaultHighWaterMark` then? Just want there to be some traceability for our decision expressed as code :)